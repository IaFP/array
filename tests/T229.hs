import Data.Array.MArray
import Data.Array.IO
import Data.Word

main :: IO ()
main = do
  -- This should fail due to integer overflow
  m <- newArray_ (0,2^62-1) :: IO (IOUArray Int Word32) -- allocates 0 bytes
  readArray m 17 >>= print -- Read some random location in address space
