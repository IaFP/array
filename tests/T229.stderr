T229: Data.Array.Base.safe_scale: Overflow
CallStack (from HasCallStack):
  error, called at libraries/array/Data/Array/Base.hs:1374:17 in array-0.5.1.2:Data.Array.Base
